nessie/lakectl_util.go

```diff
@@ -119,7 +119,9 @@ func sanitize(output string, vars map[string]string) string {
 	// The order of execution below is important as certain expression can contain others
 	// and so, should be handled first
 	s := strings.ReplaceAll(output, "\r\n", "\n")
-	s = normalizeProgramTimestamp(s)
+	if _, ok := vars["DATE"]; !ok {
+		s = normalizeProgramTimestamp(s)
+	}
```

Comment:
I think you can avoid this special case handling by running `embedVariables` **before** `sanitize` under the `update` flag in lines 145-146.
This should replace the very specific date you are expecting, with the ${DATE} placeholder before applying and replacing with the date regex.
Can you give it a quick try?

Reply:
That will help with this specific golden file creation but unfortunately not with the `fs upload` golden file, and also not with the command **result** sanitation, meaning this change will generate the correct golden file but it will never be equal to the sanitized result (line 188).

I do agree that it's not the best way to handle it and we should have a better "sanitation management".